account_financial_report/report/trial_balance.py

```diff
@@ -22,6 +22,7 @@ def _get_initial_balances_bs_ml_domain(
         date_from,
         only_posted_moves,
         show_partner_details,
+        analytic_account_id,
```

Comment:
I think this method was not well thought, as passing more and more arguments and build the domain inside is a pain for adding things like this one. Why not taking the occasion to refactor this into `_get_initial_balances_bs_ml_domain(self, extra_domain=None):` and build the extra_domain in the calling method. In fact, that domain can be used across several calls that you are also overriding.

You can do the refactor in an extra commit before the analytic account grouping option.

Reply:
Complete refactoring done. 
Now I will continue with the analytic account grouping option (in an extra commit).